### PR TITLE
BUG: Test and handle edge cases for wobbly_center

### PR DESCRIPTION
**Pull request by carterbox**
## Purpose

Adds tests which handle when the number of requested clusters is 1 or fewer and greater than the total population.

## Approach

Realized while running unrelated tests that wobbly_center was not working for all valid cluster sizes. Added tests to check all of the edge cases, then updated the docs/source accordingly.

### Submitter
- [x] Write a helpfully descriptive pull request title.
- [x] Organize changes into logically grouped commits with descriptive commit messages.
- [x] Document all new functions.
- [x] Write tests for new functions or explain why they are not needed.
- [ ] Build the documentation successfully
- [x] Use [`yapf`](https://github.com/google/yapf) to format python code.

### Reviewer
- [x] Actually read all of the code.
- [x] Run the new code yourself.
- [x] Write a summary of the changes as you understand them.
- [x] Thank the submitter.

### CODE REVIEW

File: src/tike/random.py [hunk 1]
```diff
@@ -25,31 +25,43 @@ def wobbly_center(obs, k, xp=cp):
     Parameters
     ----------
     obs : (M, N) array_like
-        The N dimensional population of N samples that needs to be clustered.
-    k : int < M
-        The number of clusters in which to divide M.
+        The N dimensional population of M samples that needs to be clustered.
+    k : int (0..M]
```
Comment (stevehenke) [inline src/tike/random.py:29]: This would be easier to read if "k" was something like "number_of_clusters" and "N" was something like "number_of_samples".

File: src/tike/random.py [hunk 2]
```diff
@@ -25,31 +25,43 @@ def wobbly_center(obs, k, xp=cp):
...
+        raise ValueError(
+            f"The number of clusters must be 0 < {k} < min(65536, M).")
     # Start with the k observations closest to the global centroid
     starting_centroids = xp.argpartition(
         xp.linalg.norm(obs - xp.mean(obs, axis=0, keepdims=True), axis=1),
         k,
         axis=0,
     )[:k]
-    # Use a label array to keep track of cluster assignment; 255 is no cluster
-    clusters = xp.empty(len(obs), dtype='uint8')
```
Comment (stevehenke) [inline src/tike/random.py:51]: Would xp.full help here?

**Review by stevehenke — APPROVED** [review]

**Comment by stevehenke** [comment]
These changes handle edge cases for the wobbly centers algorithm. The updated tests pass without issue. Thanks for your contribution!

File: src/tike/random.py (hunk 2 above)
Comment (carterbox) [reply src/tike/random.py:51]: Yes! Good idea!

File: src/tike/random.py (hunk 1 above)
Comment (carterbox) [reply src/tike/random.py]: Renamed k, to num_clusters, but left N and M alone because they are only used in the docs.
